tests/rados/test_rados_watcher_cleanup.py

```diff
@@ -0,0 +1,150 @@
+            f"Starting RADOS watcher verification for pool '{pool_name}' and image '{image_name}'"
+        )
+
+        # Step 1: Run radosbench IO for some seconds (write workload)
+        log.info("Running radosbench write workload")
+        rados_obj.run_radosbench(pg_count=8, seconds=10)
+
+        # Step 2: Verify initial watchers exist on image
+        watchers = rados_obj.get_rbd_client_ips(pool_name, image_name)
+        log.info(f"Initial watchers found: {watchers}")
+
+        if not watchers:
+            log.warning("No watchers detected initially, test may not be valid")
+
+        # Step 3: Unmount the mounted filesystem
+        log.info(f"Unmounting filesystem at {mount_point}")
+        unmount_status = unmount(rados_obj.mon, mount_point)
+        if unmount_status:
+            raise Exception(f"Failed to unmount {mount_point}")
+
+        # Step 4: Unmap the RBD device
+        log.info(f"Unmapping RBD device {device_path}")
+        rados_obj.mon.exec_command(cmd=f"rbd unmap {device_path}", sudo=True)
+
+        # Step 5: Wait for watchers to clear after unmap
+        log.info("Waiting 15 seconds for watchers to clear")
+        time.sleep(15)
+
+        watchers_after_unmap = rados_obj.get_rbd_client_ips(pool_name, image_name)
+        log.info(f"Watchers after unmap: {watchers_after_unmap}")
+
+        # Step 6: If watchers remain, blacklist those client IPs
+        if watchers_after_unmap:
+            log.info("Watchers still present, blacklisting clients to force cleanup")
+            for ip in watchers_after_unmap:
+                rados_obj.mon.exec_command(
+                    cmd=f"ceph osd blacklist add {ip}", sudo=True
+                )
+            log.info(f"Blacklisted clients: {watchers_after_unmap}")
+
+            # Wait and re-check watchers again
+            log.info("Waiting 15 seconds after blacklisting clients")
+            time.sleep(15)
+            final_watchers = rados_obj.get_rbd_client_ips(pool_name, image_name)
+            if final_watchers:
+                err_msg = f"Watchers still present after blacklisting: {final_watchers}"
+                log.error(err_msg)
+                raise Exception(err_msg)
+            else:
+                log.info("All watchers cleaned up after blacklisting")
+
+        else:
+            log.info("No watchers present after unmap, no need to blacklist")
+
+        # Step 7: Verify cluster health status at the end
+        log.info("Checking cluster health status")
+        health = rados_obj.mon.exec_command(cmd="ceph status", sudo=True)[0]
+        log.info(f"Cluster status: {health}")
+
+        return 0
+
+    except Exception as e:
+        log.error(f"An error occurred during the RADOS watcher verification: {e}")
+        raise
+
+    finally:
+        log.info(
+            "\n\n************** Execution of finally block begins here ***************\n\n"
+        )
+        log.info("Ensuring no crashes occurred during the testing")
```

Comment:
In Finally, we need to revert all the changes made in the test.

like clean-up of pool, images, removing blacklist if it exists etc.

Update the finally workflow here.